File: 23-Observer/src/go/phone_display.go
package main

import "fmt"

// PhoneDisplay is a Concrete Observer — a phone app that shows the temperature.
type PhoneDisplay struct {
	name string
}

func NewPhoneDisplay(name string) *PhoneDisplay {
	return &PhoneDisplay{name: name}
}

func (p *PhoneDisplay) Update(temperature float64) {
	fmt.Printf("  [Phone %s] It's now %.1f°C\n", p.name, temperature)
}

File: 23-Observer/src/rust/phone_display.rs
use crate::observer::Observer;

/// Concrete Observer — a phone app that shows the temperature.
pub struct PhoneDisplay {
    name: String,
}

impl PhoneDisplay {
    pub fn new(name: &str) -> Self {
        PhoneDisplay { name: name.to_string() }
    }
}

impl Observer for PhoneDisplay {
    fn update(&self, temperature: f64) {
        println!("  [Phone {}] It's now {:.1}°C", self.name, temperature);
    }
}
